backend/server.go

```diff
@@ -134,16 +135,18 @@ type serviceImpl struct {
 
 // Propose handles a replica's response to the Propose QC from the leader.
 func (impl *serviceImpl) Propose(ctx gorums.ServerCtx, proposal *hotstuffpb.Proposal) {
-	id, err := GetPeerIDFromContext(ctx, impl.srv.configuration)
-	if err != nil {
-		impl.srv.logger.Infof("Failed to get client ID: %v", err)
-		return
-	}
-
-	proposal.Block.Proposer = uint32(id)
 	proposeMsg := hotstuffpb.ProposalFromProto(proposal)
-	proposeMsg.ID = id
-
+	if !impl.srv.opts.ShouldUseKauri() {
+		id, err := GetPeerIDFromContext(ctx, impl.srv.configuration)
+		if err != nil {
+			impl.srv.logger.Infof("Failed to get client ID: %v", err)
+			return
+		}
+		proposal.Block.Proposer = uint32(id)
```

Comment:
Given that `proposal` isn't used later in this method, will this have an effect?
Or is this one still necessary: `proposeMsg := hotstuffpb.ProposalFromProto(proposal)`?

Reply:
It could perhaps be helpful to add a comment to explain why Kauri can use the ID from `proposal.Block.Proposer` while non-Kauri gets the ID from context.